frosh/devenv-meta/6.4/root/devenv.nix

```diff
@@ -0,0 +1,128 @@
+{ pkgs, config, ... }:
+
+{
+  enterShell = ''
+    rm -f .devenv/bin
+    ln -sf ${pkgs.buildEnv { name = "devenv"; paths = config.packages; ignoreCollisions = true; }}/bin .devenv/bin
+  '';
+
+  packages = [
+    pkgs.jq
+  ];
+
+  languages.javascript.enable = true;
+  languages.javascript.package = pkgs.nodejs-16_x;
+  env.NODE_OPTIONS = "--openssl-legacy-provider";
+
+  languages.php.enable = true;
+  languages.php.package = pkgs.php.buildEnv {
+    extensions = { all, enabled }: with all; enabled ++ [ amqp redis ];
+    extraConfig = ''
+      memory_limit = 2G
+      pdo_mysql.default_socket = ''${MYSQL_UNIX_PORT}
+      mysqli.default_socket = ''${MYSQL_UNIX_PORT}
+      realpath_cache_ttl = 3600
+      session.gc_probability = 0
+      session.save_handler = redis
+      session.save_path = "tcp://127.0.0.1:6379/0"
+      display_errors = On
+      error_reporting = E_ALL
+      assert.active = 0
+      opcache.memory_consumption = 256M
+      opcache.interned_strings_buffer = 20
+      zend.assertions = 0
+      short_open_tag = 0
+      zend.detect_unicode = 0
+      realpath_cache_ttl = 3600
+    '';
+  };
+
+  languages.php.fpm.pools.web = {
+    settings = {
+      "clear_env" = "no";
+      "pm" = "dynamic";
+      "pm.max_children" = 10;
+      "pm.start_servers" = 2;
+      "pm.min_spare_servers" = 1;
+      "pm.max_spare_servers" = 10;
+    };
+  };
+
+  services.caddy.enable = true;
+  services.caddy.virtualHosts.":8000" = {
+    extraConfig = ''
+      root * public
+      php_fastcgi unix/${config.languages.php.fpm.pools.web.socket}
+      encode zstd gzip
+      file_server
+      log {
+      	output stderr
+      	format console
+      	level ERROR
+      }
+    '';
+  };
+  services.caddy.virtualHosts."http://db.localhost" = {
+    extraConfig = ''
+      route * {
+        reverse_proxy localhost:8010
+      }
+    '';
+  };
+  services.caddy.virtualHosts."http://mail.localhost" = {
+    extraConfig = ''
+      route * {
+      	reverse_proxy localhost:8025
+      }
+    '';
+  };
```

Comment:
I would not add a reverse proxy to have it simple as possible and you forgot an port and it would bind to `80`